src/tasks.js

```diff
@@ -0,0 +1,123 @@
+export default class Tasks {
+  constructor() {
+    this.tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+  }
+
+  renderTasks() {
+    this.sortTasks();
+    this.updateLocalStorage();
+    const tasksContainer = document.querySelector('.tasks');
+    tasksContainer.innerHTML = '';
+    this.tasks.forEach((task) => {
+      const li = document.createElement('li');
+      li.classList.add('task');
+      li.innerHTML = `
+        <input type="checkbox" data-index="${task.index}" ${task.completed ? 'checked' : ''}>
+        <p class="task-description ${task.completed ? 'completed' : ''}">${task.description}</p>
+        <div class="delete-container" data-index="${task.index}">
+          <i class="material-icons more-vert">more_vert</i>
+          <div class="delete"><i class="material-icons">delete</i></div>
+        </div>
+      `;
+      tasksContainer.appendChild(li);
+
+      const checkboxElement = li.querySelector('input[type="checkbox"]');
+      checkboxElement.addEventListener('change', () => {
+        this.toggleTaskCompletion(task.index);
+        this.updateLocalStorage();
+        this.renderTasks();
+      });
+
+      const descriptionElement = li.querySelector('.task-description');
+      descriptionElement.addEventListener('click', () => {
+        this.editTaskDescription(descriptionElement, task.index);
+      });
+
+      const deleteIcon = li.querySelector('.delete');
+      deleteIcon.addEventListener('click', () => {
+        this.deleteTask(task.index);
+        this.updateLocalStorage();
+        this.renderTasks();
+      });
+    });
+
+    const clearCompletedElement = document.querySelector('.clear');
+    clearCompletedElement.addEventListener('click', () => {
+      this.tasks = this.tasks.filter((task) => !task.completed);
+      this.updateTaskIndexes();
+      this.updateLocalStorage();
+      this.renderTasks();
+    });
+  }
+
+  addTask(description) {
+    const task = {
+      description,
+      completed: false,
+      index: this.tasks.length + 1,
+    };
+    this.tasks.push(task);
+  }
+
+  deleteTask(index) {
+    this.tasks = this.tasks.filter((task) => task.index !== index);
+    this.updateTaskIndexes();
+  }
```

Comment:
Great design @Braingenius2 🤝

- [ ] I am not able to delete a task from your app unless I use the clear all completed button. Kindly have a look at the provided design in this [video](https://www.youtube.com/watch?v=AcUd-_Yjjqg&ab_channel=EmiliaAndrzejewska) so as to implement the delete icon properly.

Reply:
Thanks so much for your review 💯 .  @1GORDON  But please do you have any suggestions on what I can do to make it work? I have been trying to make that function work for the past 26 hours and still, no one or no solution has surfaced. Thanks so much 🙌

Reply:
- [ ] To elaborate further on the suggestion made but the previous reviewer, Kindly note that the delete function can be implemented by following a few steps: 
    - Add an event listener to the `trash-can/delete` icon that listens for a click event.
    - When the trash-can/delete icon is clicked, get the id of the todo task that needs to be deleted.
    - You can get the `id` of the todo by looping through the lists of todos you have.
    - Use the id to locate the todo task in the DOM and remove it.
    - Update the list of todos stored in localStorage to reflect the deletion of the task.
    - Update the user interface to reflect the new list of todos.

Reply:
Thanks for the review 💯 . But @Strangeal  I implemented the delete functionality and it works on my browser. When you click on the task and click on the thrash icon it deletes the task. I agree with the suggestion 🙌 I implemented all that you said already in a forEach loop when the delete/ thrash icon is created. so before the deleTask is called it already has the correct index and functions. Please look at the code. Thank you